### DISCUSSION TITLE
Gri Mech 3.0

### DISCUSSION THREAD

**Discussion by JayStandridge** [AMReX-Combustion/PeleC#685] [General]
Hi Everyone,

Does anyone have experience working with GRI mech in PeleC? I am attempting to simulate Sandia Flame-A, and I am getting spurious generation/consumption of H2O in the far field where there is nothing but air. The spontaneous consumption leads to negative concentrations of H2O, eventaully causing NaNs.

Has this been encountered by anyone else, and are there ways to fix these errors?

Thank you in advance,
Jay

**Comment by baperry2** [comment]
In the past we've had issues where negative mass fractions cause simulations to crash, but those were mostly for EB cases adjacent to the EB and those issues should be fixed. Other than that, I have not encountered anything similar to what you're describing.

Could you provide some more detail: Are you using current versions of PeleC/PelePhysics? What options are you using? is this happening near a boundary?

**Reply by JayStandridge** [reply]
I am running on the current version of PeleC and Pelephysics. I am using EB, but the issues don't appear on the EB boundary. Here is a screenshot from one of my runs:
![image](https://github.com/AMReX-Combustion/PeleC/assets/27370774/ea86509c-0cfc-4f63-94d3-830f33e48dde)
The black (with edges shown) is the EB boundary, and the spurious H2O generation/consumption and negative concentrations is shown. I am not sure what could be causing this, as the geometry works for Li and Dryer's hydrogen combustion mechanism, but the issues appear for GRI 3.0.

**Comment by drummerdoc** [comment]
Since the negative values are so small out where the reaction rates are tiny, I wonder if CVODE is taking very large internal sub cycled time steps, and this is the result of subtracting a tiny number (the production rate * dt) from an already tiny number.  With larger mechanisms, the propensity for this to happen increases.

**Reply by baperry2** [reply]
Adding to this, if using `cvode.solve_type = "GMRES"` (and perhaps some of the other options?), it may be beneficial to reduce `ode.rtol` and `ode.atol` from their default values or 1e-6 and 1e-10, respectively.

**Comment by JayStandridge** [comment]
After adjusting the refinement and ode parameters, I have a run that has negative concentrations, but manages to not blow up. 
![image](https://github.com/AMReX-Combustion/PeleC/assets/27370774/b67f2ae3-cde6-4528-af8d-0e846355fd5c)
![image](https://github.com/AMReX-Combustion/PeleC/assets/27370774/0c9e02fd-85f9-4a39-b535-5e3721ef8883)

However, I am seeing a lot of checker-boarding in my solution. I am using EB, so I am unable to use PPM to add some numerical viscosity- so how would I go about re-coupling the odd-even oscillations?

**Comment by hsitaram** [comment]
may be a basic question - with EB and MOL the cfl is limited to ~ 0.3. are
you using a high cfl?

**Reply by JayStandridge** [reply]
I am using a CFL of .3 - should I lower that?

**Reply by baperry2** [reply]
I think I've had stability issues at a CFL of 0.3 sometimes. I'd recommend trying 0.1 and then seeing how far you can increase it from there if that works.

**Reply by JayStandridge** [reply]
Will do, thank you.
